great_expectations/data_context/data_context.py

```diff
@@ -119,7 +119,7 @@
 from great_expectations.profile.basic_dataset_profiler import BasicDatasetProfiler
 from great_expectations.render.renderer.site_builder import SiteBuilder
 from great_expectations.rule_based_profiler import RuleBasedProfiler
-from great_expectations.rule_based_profiler.config import RuleBasedProfilerConfig
+from great_expectations.rule_based_profiler.types.base import RuleBasedProfilerConfig
```

Comment:
Are we sure about this? I believe we consciously decided to use `types` since it isn't informative.

I understand the value in consistency but this was a specific design decision made by @donaldheppner. Please confer with him just to get his views here.

Reply:
@cdkini We are indeed sure about this for now.  I have an agreement with @donaldheppner to introduce typed objects to `Checkpoint` and `RuleBasedProfiler` instantiation ASAP (as in -- in the soonest follow-on PRs, provided the RBP work is unblocked).  Hence, I am planning to get to it no later than the start of the next sprint (likely sooner).  Thanks.

Reply:
but if we had it in `.config` before, why are we moving it to `.types.base` now?

Reply:
@donaldheppner Please see my other response -- for now, this makes all model-type models have a "good home" in "types", hence, "config" was moved to `base.py` just like it is under the `data_context` directory organization.